.github/workflows/go.yml

```diff
@@ -369,3 +369,40 @@ jobs:
       - name: Test
         shell: bash
         run: ci/scripts/go_test.sh $(pwd)
+
+  noasm:
+    name: AMD64 Debian 11 Go ${{ matrix.go }} - NOASM
+    runs-on: ubuntu-latest
+    if: ${{ !contains(github.event.pull_request.title, 'WIP') }}
+    timeout-minutes: 60
+    strategy:
+      fail-fast: false
+      matrix:
+        go: [1.17, 1.18]
+        include:
+          - go: 1.17
+            staticcheck: v0.2.2
+          - go: 1.18
+            staticcheck: v0.3.3
+    env:
+      GO: ${{ matrix.go }}
+      STATICCHECK: ${{ matrix.staticcheck }}
+    steps:
+      - name: Checkout Arrow
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+          submodules: recursive
+      - name: Install go
+        uses: actions/setup-go@v3
+        with:
+          go-version: ${{ matrix.go }}
+          cache: true
+          cache-dependency-path: go/go.sum
+      - name: Install staticcheck
```

Comment:
side note, why isn't this one of the things archery runs? does it need to build the code to work?

Reply:
It's part of the building of the Go docker images, which is why this step doesn't exist in jobs that use the archery images. I just didn't feel like making an entirely separate docker image for building/running with the `noasm` tag.

If you look at the primary jobs which use archery, they don't have a separate `install staticcheck` step, it's in the dockerfile.

Reply:
Sure, I just mean that - is it necessary to run staticcheck separately in all pipelines, vs just once in the Archery pipeline?

Reply:
Can we not run `go_test.sh` with the noasm flag in one of the exisisting jobs instead of creating another 2 jobs? The go matrix is pretty big already :D

Reply:
@lidavidm we call staticcheck in the `go_test.sh` file so that's why I did it here, though technically we don't necessarily need it in all the pipelines but we should have it run in the both `asm` and `noasm` pipelines since it changes which files actually get included.

that said, I can definitely shoehorn this into one of the existing jobs instead as @assignUser requested :smile: I'll update this tomorrow to avoid making another job